pkg/alertmanager/operator.go

```diff
@@ -1086,6 +1096,7 @@ func (c *Operator) selectAlertmanagerConfigs(ctx context.Context, am *monitoring
 				"namespace", am.Namespace,
 				"alertmanager", am.Name,
 			)
+			c.eventRecorder.Eventf(amc, v1.EventTypeWarning, "InvalidConfiguration", "AlertmanagerConfig %s was rejected due to invalid configuration: %v", amc.GetName(), err)
```

Comment:
(nit) should we have const in pkg/operator for common reasons? e.g. `const InvalidConfiguration = "InvalidConfiguration"`.